IonDotnet.Tests/Builders/LoaderTest.cs

```diff
@@ -46,19 +56,25 @@ public void TextLoader_SymbolAnnotation()
             const string doc = "$3::123";
             var datagram = IonLoader.Default.Load(doc);
 
-            var child = (IonInt) datagram[0];
-            var annots = child.GetTypeAnnotations();
-            Assert.AreEqual(1, annots.Count);
-            Assert.AreEqual(SystemSymbols.IonSymbolTable, annots.First().Text);
+            foreach (var ionValue in datagram)
```

Comment:
There should only be one element, so I'd assert that there's only one and then index element zero instead of looping.

Reply:
We cannot access elements via indexing since ``IIonValue`` is an interface, hence looping. We get ``Cannot apply indexing to type IIonValue `` error.

Added assertion for number of elements.

Reply:
You can use `datagram.IndexOf`, correct?

Reply:
I see we have this in `IonSequence`: 
```
public IIonValue this[int index] {
...
}
```
Would moving this declaration into `IIonSequence` allow us to index? If not, we should probably add a method to `IIonSequence` that allows the user to get the value at a particular index, even if not via `[]` syntax.

Reply:
Done.